server/sonar-server/src/main/java/org/sonar/server/rule/index/RuleIndex.java

```diff
@@ -389,66 +410,41 @@ private void addActiveSeverityFacetIfNeeded(RuleQuery query, QueryContext option
-    }
-    if (options.facets().contains(FACET_TAGS) || options.facets().contains(FACET_OLD_DEFAULT)) {
-      Collection<String> tags = query.getTags();
-      aggregations.put(FACET_TAGS,
-        stickyFacetBuilder.buildStickyFacet(RuleNormalizer.RuleField.ALL_TAGS.field(), FACET_TAGS,
-          tags == null ? new String[0] : tags.toArray()));
-    }
-    if (options.facets().contains("repositories") || options.facets().contains(FACET_OLD_DEFAULT)) {
-      Collection<String> repositories = query.getRepositories();
-      aggregations.put(FACET_REPOSITORIES,
-        stickyFacetBuilder.buildStickyFacet(RuleNormalizer.RuleField.REPOSITORY.field(), FACET_REPOSITORIES,
-          repositories == null ? new String[0] : repositories.toArray()));
-    }
+  private StickyFacetBuilder stickyFacetBuilder(QueryBuilder query, Map<String, FilterBuilder> filters) {
+    return new StickyFacetBuilder(query, filters);
   }
 
-  public Result<Rule> search(RuleQuery query, QueryContext options) {
-    SearchRequestBuilder esSearch = getClient()
-      .prepareSearch(this.getIndexName())
-      .setTypes(this.getIndexType())
-      .setIndices(this.getIndexName());
-
-    if (options.isScroll()) {
-      esSearch.setSearchType(SearchType.SCAN);
-      esSearch.setScroll(TimeValue.timeValueMinutes(3));
-    }
-
-    QueryBuilder qb = this.getQuery(query);
-    Map<String, FilterBuilder> filters = this.getFilters(query, options);
-
-    if (options.isFacet()) {
-      for (AggregationBuilder aggregation : getFacets(query, options, qb, filters).values()) {
-        esSearch.addAggregation(aggregation);
+  private void setSorting(RuleQuery query, SearchRequestBuilder esSearch) {
+    /* integrate Query Sort */
+    String queryText = query.getQueryText();
+    if (query.getSortField() != null) {
+      FieldSortBuilder sort = SortBuilders.fieldSort(appendSortSuffixIfNeeded(query.getSortField()));
+      if (query.isAscendingSort()) {
+        sort.order(SortOrder.ASC);
+      } else {
+        sort.order(SortOrder.DESC);
       }
+      esSearch.addSort(sort);
+    } else if (queryText != null && !queryText.isEmpty()) {
+      esSearch.addSort(SortBuilders.scoreSort());
+    } else {
+      esSearch.addSort(appendSortSuffixIfNeeded(RuleIndexDefinition.FIELD_RULE_UPDATED_AT), SortOrder.DESC);
+      // deterministic sort when exactly the same updated_at (same millisecond)
+      esSearch.addSort(appendSortSuffixIfNeeded(RuleIndexDefinition.FIELD_RULE_KEY), SortOrder.ASC);
     }
+  }
 
-    setSorting(query, esSearch);
-    setPagination(options, esSearch);
-    setFields(options, esSearch);
-
-    BoolFilterBuilder fb = FilterBuilders.boolFilter();
-    for (FilterBuilder ffb : filters.values()) {
-      fb.must(ffb);
-    }
-
-    esSearch.setQuery(QueryBuilders.filteredQuery(qb, fb));
-    SearchResponse esResult = esSearch.get();
-    return new Result<>(this, esResult);
+  public static String appendSortSuffixIfNeeded(String field) {
+      return field +
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AIndentationCheck)